### PR TITLE
remove CrossChainApplications examples

### PR DESCRIPTION
**Pull request by feuGeneA**
## Why this should be merged
Fixes https://github.com/ava-labs/teleporter/issues/381

The actual migration of any of these examples, into the avalanche-starter-kit, as specified by https://github.com/ava-labs/avalanche-starter-kit/issues/5, can happen independently of this removal.

## How this was tested
I found that you can enable a go coverage report for Ginkgo test runs. Of course it doesn't provide any coverage for the contracts themselves, but I was able to focus the coverage report on just the contracts' ABI bindings. I found that the coverage report was identical before and after the removal of these tests. Therefore, I think removing these tests likely has no impact on test coverage.

### CODE REVIEW

**Review by geoff-vball — APPROVED** [review]

**Review by michaelkaplan13 — COMMENTED** [review]
Can you add a prominent reference to the starter kit repo from the top level README for people looking to learn how to build applications on top of Teleporter?

**Comment by cam-schultz** [comment]
> Of course it doesn't provide any coverage for the contracts themselves, but I was able to focus the coverage report on just the contracts' ABI bindings

This should be an accurate view of the e2e contract code coverage (at least when comparing before/after), since we only ever interact with the contracts via the ABI bindings in the tests.

**Review by geoff-vball — APPROVED** [review]

**Review by cam-schultz — APPROVED** [review]

File: contracts/src/Teleporter/tests/TestMessenger.sol [hunk 1]
```diff
@@ -17,10 +17,10 @@ import {ReentrancyGuard} from "@openzeppelin/contracts@4.8.1/security/Reentrancy
  */
 
 /**
- * @dev ExampleCrossChainMessenger is an example contract that demonstrates how to send and receive
+ * @dev TestMessenger is test fixture contract that exercises sending and receiving
```
Comment (bernard-avalabs) [inline contracts/src/Teleporter/tests/TestMessenger.sol:20]:
```suggestion
 * @dev TestMessenger is a test fixture contract that exercises sending and receiving
```

File: contracts/src/Teleporter/tests/TestMessenger.sol [hunk 2]
```diff
@@ -17,10 +17,10 @@ import {ReentrancyGuard} from "@openzeppelin/contracts@4.8.1/security/Reentrancy
...
  * messages cross chain.
```
Comment (bernard-avalabs) [inline contracts/src/Teleporter/tests/TestMessenger.sol:21]:
```suggestion
 * messages across chain.
```

**Review by bernard-avalabs — APPROVED** [review]
LGTM. Just pointed out some minor typos.

**Pull request merged**